### PR TITLE
fix: delay SDK shutdown until proxy has signalled that it terminated

### PR DESCRIPTION
**Pull request by johanandren**
Refs #21

Draft until discovery protocol change is available https://github.com/lightbend/akkaserverless-framework/pull/714

### CODE REVIEW

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
For the record I manually verified it by running against current master with the protocol change in place

File: sdk/src/akkaserverless.js [hunk 1]
```diff
@@ -183,13 +187,24 @@ class AkkaServerless {
...
     this.server.start();
     console.log("gRPC server started on " + opts.bindAddress + ":" + boundPort);
 
+    process.on('SIGTERM', function onSigterm () {
+      if (!this.proxySeen || this.proxyHasTerminated || this.devMode) {
+        debug('Got SIGTERM. Shutting down')
+        this.terminate()
+      } else {
+        debug('Got SIGTERM. But did not yet see proxy terminating, deferring shutdown until proxy stops')
+        // no timeout because process will be SIGKILLed anyway if it does not get the proxy termination in time
```
Comment (pvlugter) [inline sdk/src/akkaserverless.js:204]: Was wondering about a timeout, but makes sense that it will be killed eventually anyway.

**Review by pvlugter — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
framework `0.7.0-beta.10` has been released

**Pull request merged**